horovod/run/driver/driver_service.py

```diff
@@ -103,16 +103,27 @@ def _exec_command(command):
                         driver_addresses=codec.dumps_base64(driver_addresses),
                         settings=codec.dumps_base64(settings))
         else:
-            command = \
-                'ssh -o StrictHostKeyChecking=no {host} {ssh_port_arg} ' \
-                '\'{python} -m horovod.run.task_fn {index} {driver_addresses}' \
-                ' {settings}\''\
-                .format(host=host_name,
-                        ssh_port_arg=ssh_port_arg,
-                        python=sys.executable,
-                        index=codec.dumps_base64(index),
-                        driver_addresses=codec.dumps_base64(driver_addresses),
-                        settings=codec.dumps_base64(settings))
+            if env_util.is_mpi_job():
```

Comment:
It looks like there's an opportunity to make this more generic and reduce code duplication.

First, I would suggest adding a helper function to `env.py`:

```
def get_remote_command(local_command, host, ssh_port_arg=None):
    if is_kubeflow_mpi():
        return f'/etc/mpi/kubexec.sh {host} {local_command}'
    else:
        ssh_port_arg = ssh_port_arg or ''
        return f'ssh -o StrictHostKeyChecking=no {host} {ssh_port_arg} {local_command}'
```

Then you can refactor this and the other sections of code:

```
local_command = '\'{python} -m horovod.run.task_fn {index} {driver_addresses}' \
                    ' {settings}\'' \
                        .format(
                                python=sys.executable,
                                index=codec.dumps_base64(index),
                                driver_addresses=codec.dumps_base64(driver_addresses),
                                settings=codec.dumps_base64(settings))
command = get_remote_command(command, host_name, ssh_port_arg)
```

What do you think?

Reply:
I noticed that on the master branch, `get_ssh_command` is defined under `horovod/runner/util/remote.py`. We can further reduce the duplication by adding a `get_remote_command` there. It also reduce the modification to the rest of scripts.

Please take a look at the implementation I just pushed.

Reply:
Thanks!  Yes, I just implemented `get_ssh_command` yesterday for a different issue, and thought it would help to go ahead and design it in a way the fit into this PR.  I'll take a look.